proof/invariant-abstract/AARCH64/ArchTcbAcc_AI.thy

```diff
@@ -175,4 +169,110 @@ global_interpretation TcbAcc_AI?: TcbAcc_AI
   case 1 show ?case by (unfold_locales; (fact TcbAcc_AI_assms)?)
   qed
 
+context Arch begin arch_global_naming
+
+\<comment> \<open>FIXME: do we want a generic version with valid_arch_idle in the precondition\<close>
+lemma arch_thread_set_valid_idle[wp]:
+  "\<lbrace>valid_idle and (\<lambda>s. t \<noteq> idle_thread s \<or> (\<forall>atcb. tcb_vcpu atcb = None \<longrightarrow> tcb_vcpu (f atcb) = None))\<rbrace>
+   arch_thread_set f t
+   \<lbrace>\<lambda>rv. valid_idle\<rbrace>"
```

Comment:
After completing the proof, do you think it would be useful? I guess we'll know more about the arch interface when we have made the other architectures work again.

Reply:
I've pulled this commit into a separate [PR ](https://github.com/seL4/l4v/pull/852)so that I could do all of the architectures now. Doing that convinced me that I might as well make the generic version and then more specific arch ones as needed.